src/Nethermind/Nethermind.TxPool/Collections/DistinctValueSortedPool.cs

```diff
@@ -16,47 +16,53 @@
 
 using System;
 using System.Collections.Generic;
+using System.Diagnostics.CodeAnalysis;
 
 namespace Nethermind.TxPool.Collections
 {
-    public class DistinctValueSortedPool<TKey, TValue> : SortedPool<TKey, TValue>
+    public class DistinctValueSortedPool<TKey, TValue, TGroup> : SortedPool<TKey, TValue, TGroup>
     {
-        private readonly IDictionary<TValue, LinkedListNode<KeyValuePair<TKey, TValue>>> _distinctDictionary;
+        private readonly IComparer<TValue> _comparer;
+        private readonly IDictionary<TValue, KeyValuePair<TKey, TValue>> _distinctDictionary;
 
-        public DistinctValueSortedPool(int capacity, Comparison<TValue> comparison, IEqualityComparer<TValue> distinctComparer) : base(capacity, comparison)
+        public DistinctValueSortedPool(
+            int capacity,
+            IComparer<TValue> comparerWithIdentity,
```

Comment:
what is comparer with identity?

Reply:
comparer that will compare based on sorting, and in the end if sorting comparision is 0 compare based on identity. Needed to keep all entities in SortedDictionary, otherwise if they are equal it can only keep first one.